lib/spack/spack/rewiring.py

```diff
@@ -113,22 +98,18 @@ def rewire_node(spec, explicit):
                 spec.prefix,
             )
         relocate.relocate_text_bin(binaries=bins_to_relocate, prefixes=prefix_to_prefix)
-    # Copy package into place, except for spec.json (because spec.json
-    # describes the old spec and not the new spliced spec).
-    shutil.copytree(
-        os.path.join(tempdir, spec.dag_hash()),
+    shutil.rmtree(tempdir)
```

Comment:
(minor, just a comment, possibly for later) Here we can consider using `tempfile.TemporaryDirectory` as a context manager

Reply:
yeah, it would be better to merge this with the other binary relocation code, see #48262